packages/dev/core/src/Materials/Textures/multiRenderTarget.ts

```diff
@@ -290,6 +317,37 @@ export class MultiRenderTarget extends RenderTargetTexture {
         if (this._multiRenderTargetOptions.useSRGBBuffers) {
             this._multiRenderTargetOptions.useSRGBBuffers[index] = texture._useSRGBBuffer;
         }
+        if (this._multiRenderTargetOptions.targetTypes) {
+            let target: number = 0;
+            if (texture.is2DArray) {
+                target = Constants.TEXTURE_2D_ARRAY;
+            } else if (texture.isCube) {
+                target = Constants.TEXTURE_CUBE_MAP;
+            } else if (texture.is3D) {
+                target = Constants.TEXTURE_3D;
+            } else {
+                target = Constants.TEXTURE_2D;
+            }
+            this._multiRenderTargetOptions.targetTypes[index] = target;
+        }
+    }
+
+    /**
+     * Changes an attached texture's face index or layer.
+     * @param faceIndexOrLayer The face index or layer of the texture to be attached
+     * @param index The index of the texture to modify the attachment of
+     */
+    public setInternalFaceIndexOrLayer(faceIndexOrLayer: number, index: number) {
+        if (!this.textures[index]) {
+            return;
+        }
+
+        if (this._multiRenderTargetOptions.faceIndexOrLayer) {
+            this._multiRenderTargetOptions.faceIndexOrLayer[index] = faceIndexOrLayer;
+        }
+
+        const renderTarget = this.renderTarget as WebGLRenderTargetWrapper;
+        renderTarget.setTexture(this.textures[index].getInternalTexture()!, index, false, faceIndexOrLayer);

```

Comment:
Replace these two lines with a call to `this.renderTarget.setLayerAndFaceIndex()`.